zebra-chain/src/block/tests/generate.rs

```diff
@@ -125,8 +226,112 @@ fn single_transaction_block(oversized: bool) -> Block {
 
     // Put the big transaction into a block
     let transactions = vec![Arc::new(big_transaction)];
-    Block {
-        header,
+
+    let block = Block {
+        header: block_header,
         transactions,
+    };
+
+    let serialized_len = block.zcash_serialize_to_vec().unwrap().len();
+    assert_eq!(
+        oversized,
+        serialized_len > MAX_BLOCK_BYTES.try_into().unwrap(),
+        "block is over-sized if requested:\n\
+         oversized: {},\n\
+         serialized_len: {},\n\
+         MAX_BLOCK_BYTES: {},",
+        oversized,
+        serialized_len,
+        MAX_BLOCK_BYTES,
```

Comment:
Suggestion: small tweak for conciseness.

```suggestion
         oversized: {oversized},\n\
         serialized_len: {serialized_len},\n\
         MAX_BLOCK_BYTES: {MAX_BLOCK_BYTES},",
```